<launch>
  <!-- configure console output verbosity mode:debug_console.conf or std_console.conf -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find spinnaker_sdk_camera_driver)/cfg/debug_console.conf"/>

  <!-- acquisition spinnaker params-->
  <arg name="output"            default="screen"	doc="display output to screen or log file"/>
  <arg name="time"              default="false"	doc="Show time/FPS on output"/>
  <arg name="utstamps"          default="false"	doc="Flag whether each image should have Unique timestamps vs the master cams time stamp for all" />
  <arg name="config_file"       default="$(find spinnaker_sdk_camera_driver)/params/stereo.yaml" doc="File specifying the parameters of the camera_array"/>

  <!-- nodelet params-->
  <arg name="nodelet_manager_name" default="vision_nodelet_manager" doc="name of the nodelet manager, comes handy when launching multiple nodelets from different launch files" />
  <arg name="start_nodelet_manager" default="true" doc="If set to True(default), creates a nodelet manager with $(arg nodelet_manager_name).
  If False, the acquisition/capture_nodelet waits for the nodelet_manager name $(arg nodelet_manager_name)" />

   <!-- start the nodelet manager if $(arg start_nodelet_manager) is true-->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen" if="$(arg start_nodelet_manager)" />

  <!-- load the acquisition nodelet -->
  <node pkg="nodelet" type="nodelet" name="acquisition_node"
          args="load acquisition/capture_nodelet $(arg nodelet_manager_name)" >
    <!-- load the acquisition node parameters file.  Note any parameters provided in this file will 
	 override what is in the yaml file.  Thus use it to set parameters camer_array configuration params -->
    <rosparam command="load"        file="$(arg config_file)" />

    <!-- Load parameters onto server using argument or default values above -->
    <param name="frame_id"          value="left_cam_optical" />
    <param name="time"              value="$(arg time)" />
    <param name="utstamps"          value="$(arg utstamps)" />
    <param name="camera_info_root_url" value="file://$(find max_vision)/calibration/" />
    <param name="hardware_id"          value="stereo_cam" />

  </node>

</launch>
